### PR TITLE
Update sandiatoss3 pe-layouts for failing tri-grid tests

### PR DESCRIPTION
**Pull request by jonbob**
This PR adds two specific pe-layouts for sandiatoss3 to run two failing tri-grid tests:
* SMS.r05_r05.I1850CLM45CN.sandiatoss3_intel
* SMS_D_Ld1.ne30_r05_oECv3.A_WCYCL1850.sandiatoss3_intel
These layouts were tested in [SES-875](https://acme-climate.atlassian.net/servicedesk/customer/portal/2/SES-875) and now pass.

[BFB] for all previously tested configurations

### CODE REVIEW

**Comment by wadeburgess** [comment]
I don't think I should review this. I am unfamiliar with pe-layouts.

File: cime/config/e3sm/allactive/config_pesall.xml [hunk 1]
```diff
@@ -1276,6 +1276,80 @@
...
+          <rootpe_ocn>0</rootpe_ocn>
+          <rootpe_glc>0</rootpe_glc>
+          <rootpe_wav>0</rootpe_wav>
+          <rootpe_cpl>0</rootpe_cpl>
+        </rootpe>
+      </pes>
+    </mach>
+  </grid>
+  <grid name="a%r05_l%r05_oi%null_r%r05_g%null_w%null_z%null_m%oEC60to30v3">
+    <mach name="sandiatoss3">
```
Comment (amametjanov) [inline cime/config/e3sm/allactive/config_pesall.xml]: I think it'd be good to set `<mach name="any">` for both grids, tests with these grids are running into out-of-memory errors on multiple machines.

Comment (jonbob) [reply cime/config/e3sm/allactive/config_pesall.xml]: Thanks for looking, @amametjanov . Do all the machines we test on have the resources? I can't remember how big melvin is...

Comment (amametjanov) [reply cime/config/e3sm/allactive/config_pesall.xml]: @wadeburgess can you try
```
./cime/scripts/create_test SMS_P256.r05_r05.I1850CLM45CN SMS_D_Ld1_P256.ne30_r05_oECv3.A_WCYCL1850
```
on Melvin?

Comment (jonbob) [reply cime/config/e3sm/allactive/config_pesall.xml]: SMS.r05_r05.I1850CLM45CN.melvin_gnu passes without changing the layout

Comment (amametjanov) [reply cime/config/e3sm/allactive/config_pesall.xml]: IIRC, Melvin is a single node with 48 cores and very large memory. Oversubscribing (256 tasks on 48 cores) is allowed.

Another option is to set
```
<mach name="sandiatoss3|cori-knl|cori-haswell|theta|anvil|bebop">
```

Comment (jgfouca) [reply cime/config/e3sm/allactive/config_pesall.xml]: Melvin has a ton of memory but not 256 cores (we oversubscribe up to 64 on melvin)

Comment (jonbob) [reply cime/config/e3sm/allactive/config_pesall.xml]: Thanks for all your help, @amametjanov and @jgfouca. I will change it to Az's second option, listing all the machine names...

**Comment by jonbob** [comment]
merged to next

**Comment by jonbob** [comment]
merged to master

**Pull request merged**